src/kubernetes.cc

```diff
@@ -1306,15 +1028,12 @@ void KubernetesUpdater::StartUpdater() {
     pod_watch_thread_ = std::thread([=]() {
       reader_.WatchPods(watched_node, cb);
     });
-    if (config().KubernetesClusterLevelMetadata() &&
-        config().KubernetesServiceMetadata()) {
-      service_watch_thread_ = std::thread([=]() {
-        reader_.WatchServices(cb);
-      });
-      endpoints_watch_thread_ = std::thread([=]() {
-        reader_.WatchEndpoints(cb);
-      });
-    }
+    service_watch_thread_ = std::thread([=]() {
```

Comment:
These are still cluster-level, aren't they? Are you going to have every node agent watch for services and endpoints?
I would also prefer to keep the `KubernetesServiceMetadata` option, at least for now.

Reply:
Yes - they are still cluster-level. Added back the two checks.

What does keeping the `KubernetesServiceMetadata` option for now mean? Would it only apply to services and endpoints? Or other cluster level resources too?